repose-aggregator/components/filters/rackspace-auth-user-filter/src/main/scala/org/openrepose/filters/rackspaceauthuser/RackspaceAuthUserFilter.scala

```diff
@@ -75,65 +77,50 @@ class RackspaceAuthUserFilter @Inject()(configurationService: ConfigurationServi
 
       wrappedResponse.getHeadersList(CommonHttpHeader.WWW_AUTHENTICATE.toString).asScala.filter(_.startsWith("OS-MF")) foreach { header =>
         Option(StringUtils.substringBetween(header, "sessionId='", "'")) match {
-          case Some(sessionId) =>
-            datastore.put(s"$ddKey:$sessionId", authUserGroup, 5, TimeUnit.MINUTES)
-          case None =>
-            logger.debug("Failed to parse the session id out of '{}'", header)
+          case Some(sessionId) => datastore.put(s"$ddKey:$sessionId", authUserGroup, 5, TimeUnit.MINUTES)
+          case None => logger.debug("Failed to parse the session id out of '{}'", header)
         }
       }
     }
   }
 
-  type UsernameParsingFunction = InputStream => (Option[String], Option[String])
-  val username1_1XML: UsernameParsingFunction = { is =>
+  val usernameAuth1_1Xml: UsernameParsingFunction = { is =>
     val xml = XML.load(is)
     val username = (xml \\ "credentials" \ "@username").text
-    if (username.nonEmpty) {
-      (None, Some(username))
-    } else {
-      (None, None)
-    }
+    (None, Option(username).filterNot(_.isEmpty))
   }
   // https://www.playframework.com/documentation/2.3.x/ScalaJson
   //Using play json here because I don't have to build entire objects
-  val username1_1JSON: UsernameParsingFunction = { is =>
+  val usernameAuth1_1Json: UsernameParsingFunction = { is =>
     val json = Json.parse(is)
     val username = (json \ "credentials" \ "username").validate[String]
     username match {
       case s: JsSuccess[String] =>
         (None, Some(s.get))
       case f: JsError =>
-        logger.debug(s"1.1 JSON parsing failure: ${
-          JsError.toFlatJson(f)
-        }")
+        logger.debug(s"1.1 JSON parsing failure: ${JsError.toFlatJson(f)}")
         (None, None)
     }
   }
 
-  def getUsername(domain: String, user: String): String = {
-    if (domain.equals("Rackspace")) {
-      "Racker:" + user
-    } else {
-      user
-    }
-  }
+  def getUsername(domain: String, user: String): String = if (domain.equals("Rackspace")) s"Racker:$user" else user
```

Comment:
Not sure i like the collapse here. Probably because i don't like the lack of a ternary operator in scala. This is one of those things that is a bug waiting to happen.